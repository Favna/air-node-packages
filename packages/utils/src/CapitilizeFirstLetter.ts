/**
 * Transforms the first letter to a capital then adds all the rest after it
 * Differs from {@link sentenceCase} in that the casing of the text after the first letter is retained
 * @param str Text to transform
 */
export function capitilizeFirstLetter(str: string) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}
