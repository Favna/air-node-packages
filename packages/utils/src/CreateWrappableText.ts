/**
 * Inserts a space in any text after every maxLength amount of characters, allowing Material-UI to gracefully wrap it
 * @param text text to split
 * @param maxLength The maximum length of a text section @default 75
 */
export function createWrappableText(text: string, maxLength = 75) {
  const splitRegex = new RegExp(`.{1,${maxLength}}`, 'g');
  const splitContent = text.match(splitRegex);

  if (splitContent) return splitContent.join(' ');

  return text;
}

export default createWrappableText;
