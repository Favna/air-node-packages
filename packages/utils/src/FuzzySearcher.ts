import Fuse from 'fuse.js';

export class FuzzySearcher<O> {
  private readonly object: O[];
  private readonly fuseOptions?: Fuse.IFuseOptions<unknown>;

  constructor(obj: O[], keys: string[], options?: Fuse.IFuseOptions<unknown>) {
    this.object = obj;
    this.fuseOptions = {
      threshold: 0.2,
      ...options,
      keys
    };
  }

  public runFuzzy(query: string) {
    const locquery = query.toLowerCase();

    const fuzzyFuze = new Fuse(this.object, this.fuseOptions);

    return fuzzyFuze.search<O>(locquery);
  }
}
