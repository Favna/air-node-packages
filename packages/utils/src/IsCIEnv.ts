/** Validates whether the current environment is a CI environment */
export const isCIEnv = Boolean(process.env.CI);

export default isCIEnv;