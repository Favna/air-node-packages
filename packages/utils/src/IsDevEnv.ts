/** Validates whether the current environment is either development or test */
export const isDevEnv = process.env.NODE_ENV === 'development' ? true : process.env.NODE_ENV === 'test';

export default isDevEnv;
