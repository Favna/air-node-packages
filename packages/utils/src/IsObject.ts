/**
 * Verify if the input is an object literal (or class).
 * @param input The object to verify
 */
export const isObject = (input: unknown): boolean => {
  return input ? (input as object).constructor === Object : false;
};

export default isObject;