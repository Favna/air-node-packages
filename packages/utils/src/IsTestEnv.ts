/** Validates whether the current environment is test */
export const isTestEnv = process.env.NODE_ENV === 'test';

export default isTestEnv;
