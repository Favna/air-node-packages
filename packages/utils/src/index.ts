export * from './CapitilizeFirstLetter';
export * from './CreateWrappableText';
export * from './CutText';
export * from './IsDevEnv';
export * from './IsTestEnv';
export * from './ObjectHasProperty';
export * from './SentenceCase';
export * from './SnakeToCamelCase';
export * from './SplitText';
export * from './StableTableSort';
export * from './TimeUtils';
export * from './UseWindowSize';