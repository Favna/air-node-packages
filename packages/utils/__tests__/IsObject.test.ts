import { isObject } from '../src';

describe('isObject', () => {
  test('isObject(string)', () => {
    const value = 'Hello World';
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(number)', () => {
    const value = 420;
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(bigint)', () => {
    // eslint-disable-next-line no-undef
    const value = BigInt(420);
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(boolean)', () => {
    const value = true;
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(undefined)', () => {
    const value = undefined;
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(object)', () => {
    const value = { class: '' };
    expect(isObject(value)).toBe(true);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(object-null)', () => {
    const value = null;
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(object-array)', () => {
    const value: unknown[] = [];
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(object-non-literal)', () => {
    // eslint-disable-next-line
    const value = new (class A {})();
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(function)', () => {
    // eslint-disable-next-line
    const value = function myClass(): void {
      /* Noop */
    };
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(arrow)', () => {
    const value = (): void => {
      /* Noop */
    };
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('isObject(class)', () => {
    // eslint-disable-next-line
    const value = class A {};
    expect(isObject(value)).toBe(false);
    expect(isObject(value)).toMatchSnapshot();
  });

  test('Function Snapshot', () => {
    expect(isObject.toString()).toMatchSnapshot();
  });
});
