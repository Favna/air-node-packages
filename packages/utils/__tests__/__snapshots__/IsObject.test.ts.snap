// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isObject Function Snapshot 1`] = `
"function (input) {
    return input ? input.constructor === Object : false;
}"
`;

exports[`isObject isObject(arrow) 1`] = `false`;

exports[`isObject isObject(bigint) 1`] = `false`;

exports[`isObject isObject(boolean) 1`] = `false`;

exports[`isObject isObject(class) 1`] = `false`;

exports[`isObject isObject(function) 1`] = `false`;

exports[`isObject isObject(number) 1`] = `false`;

exports[`isObject isObject(object) 1`] = `true`;

exports[`isObject isObject(object-array) 1`] = `false`;

exports[`isObject isObject(object-non-literal) 1`] = `false`;

exports[`isObject isObject(object-null) 1`] = `false`;

exports[`isObject isObject(string) 1`] = `false`;

exports[`isObject isObject(undefined) 1`] = `false`;
